tests/github_client/mod.rs

```diff
@@ -0,0 +1,876 @@
+//! `GithubClient` tests.
+//!
+//! These tests exercise the behavior of `GithubClient`. They involve setting
+//! up HTTP servers, creating a `GithubClient` to connect to those servers,
+//! executing some action, and validating the result.
+//!
+//! The [`run_test`] function is used to set up the test and give you a
+//! [`GithubClient`] to perform some action and validate its result.
+//!
+//! To write one of these tests, you'll need to use the recording function
+//! against the live GitHub site to fetch what the actual JSON objects should
+//! look like. To write a test, follow these steps:
+//!
+//! 1. Choose a repo in your GitHub account to run tests against. I use either
+//!    `ehuss/rust` or `ehuss/triagebot-test`, but any in your account will do.
```

Comment:
Am I right in interpreting this means that if (for example) I adjust the functionality for some test -- passing different parameters, querying slightly different information, etc. -- I have no good way of re-running it without changing lots of irrelevant details in the output? (Since I lack access to your repository for test creation etc.).

I think the tradeoff probably still makes sense - in practice hopefully we can avoid updating these files too much -- but would like to hear your take on that.

Reply:
a variant of the above comment: is there a way to regenerate the JSON files automatically - in case of said small change? maybe I think no, right? You have to delete all of the relevant JSON test files, make said small change, re-run the tests and try to `diff` the new JSON generated files?

Reply:
Yea, regenerating an existing test is not particularly easy. My expectation is that the process would look like:

1. Make a change to `GithubClient`.
2. Edit the test to point to a repo you control (like `Mark-Simulacrum/rust`).
3. Re-record the test using `TRIAGEBOT_TEST_RECORD_DIR=github_client/TEST_NAME_HERE cargo test --test testsuite -- --exact github_client::TEST_NAME_HERE`
4. Add any new assertions relevant to your change (or fix any errors with the existing assertions).

Unfortunately some of the tests have setup that is required (like creating a PR, creating labels, etc.) that the tests currently don't do (I did that manually). I can look at making those tests do all the setup they need. #1678 does this to some degree, but it requires more code to do the setup.

Regenerating all of the JSON files isn't currently possible. The tests need the ability to make changes on the live GitHub site, which means it needs to have a repo that it can write to, and it doesn't know what that is. Per the comment below, I can look at adding that possibility. But that would end up changing all the JSON files causing a large diff.